File: app/src/main/java/com/my/televip/virtuals/ui/Cells/TextCheckCell.java
package com.my.televip.virtuals.ui.Cells;

import android.content.Context;
import android.view.View;
import android.widget.TextView;

import com.my.televip.Class.ClassLoad;
import com.my.televip.Class.ClassNames;
import com.my.televip.obfuscate.Obfuscate;

import de.robv.android.xposed.XposedHelpers;

public class TextCheckCell {

    public Object textCell;

    public TextCheckCell(Context context){
        textCell = XposedHelpers.newInstance(ClassLoad.getClass(ClassNames.TEXT_CHECK_CELL), context);
    }

    public TextCheckCell(Object obj){
        textCell = obj;
    }

    public void setTextAndValueAndCheck(CharSequence text, String value, boolean checked, boolean multiline, boolean divider){
        XposedHelpers.callMethod(textCell, Obfuscate.getMethodName("TextCheckCell","setTextAndValueAndCheck"), text, value, checked, multiline,  divider);
    }

    public void setTextAndCheck(CharSequence text, boolean checked, boolean divider){
        XposedHelpers.callMethod(textCell, Obfuscate.getMethodName("TextCheckCell","setTextAndCheck"), text, checked,  divider);
    }

    public void setChecked(boolean checked){
        XposedHelpers.callMethod(textCell, Obfuscate.getMethodName("TextCheckCell","setChecked"), checked);
    }

    public boolean isChecked(){
        return (boolean) XposedHelpers.callMethod(textCell, Obfuscate.getMethodName("TextCheckCell","isChecked"));
    }

    public TextView getTextView(){
        return (TextView) XposedHelpers.getObjectField(textCell, Obfuscate.getFieldName("TextCheckCell","textView"));
    }

    public View getView(){
        return (View) textCell;
    }

}

File: app/src/main/java/com/my/televip/virtuals/ui/Cells/TextCheckCell.kt
package com.my.televip.virtuals.ui.Cells

import android.content.Context
import android.view.View
import android.widget.TextView
import com.my.televip.clazz.ClassLoad
import com.my.televip.clazz.ClassNames
import com.my.televip.obfuscate.Obfuscate
import de.robv.android.xposed.XposedHelpers

class TextCheckCell {

    @JvmField
    var textCell: Any? = null

    constructor(context: Context?) {
        val tcClass = ClassLoad.getClass(ClassNames.TEXT_CHECK_CELL)
        textCell = XposedHelpers.newInstance(tcClass, context)
    }

    constructor(obj: Any?) {
        textCell = obj
    }

    fun setTextAndValueAndCheck(text: CharSequence?, value: String?, checked: Boolean, multiline: Boolean, divider: Boolean) {
        val target = textCell ?: return
        XposedHelpers.callMethod(target, Obfuscate.getMethodName("TextCheckCell", "setTextAndValueAndCheck"), text, value, checked, multiline, divider)
    }

    fun setTextAndCheck(text: CharSequence?, checked: Boolean, divider: Boolean) {
        val target = textCell ?: return
        XposedHelpers.callMethod(target, Obfuscate.getMethodName("TextCheckCell", "setTextAndCheck"), text, checked, divider)
    }

    fun setChecked(checked: Boolean) {
        val target = textCell ?: return
        XposedHelpers.callMethod(target, Obfuscate.getMethodName("TextCheckCell", "setChecked"), checked)
    }

    val isChecked: Boolean
        get() {
            val target = textCell ?: return false
            return XposedHelpers.callMethod(target, Obfuscate.getMethodName("TextCheckCell", "isChecked")) as? Boolean ?: false
        }

    val textView: TextView?
        get() {
            val target = textCell ?: return null
            return XposedHelpers.getObjectField(target, Obfuscate.getFieldName("TextCheckCell", "textView")) as? TextView
        }

    val view: View?
        get() = textCell as? View
}
